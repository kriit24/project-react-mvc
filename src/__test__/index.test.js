import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Project, App, Loader, Popup} from "../index";
import renderer from 'react-test-renderer';
import expect from "expect";
import ControllerIndex from './app/controller/index';

export default class MainApp extends Project.React {

    state = {
        'load': true,
        'reload' : null
    };

    constructor() {

        super();
        //let makes async tasks before continue
        //import translates from server
        //get user data from server
        let promise = [];

        promise.push(new Promise(async (resolve, reject) => {
            Popup.init({
                confirm: {
                    YES: 'YES',
                    NO: 'NO',
                }
            });
            resolve('done');
        }));

        Promise.all(promise).then(() => {

            this.setState({'load' : false});
        }).catch((error) => {

            this.setState({'load' : false});
        });

        //set globally App.Reload() AND App.User()
        App.export({
            Reload: () => {

                this.setState({'reload' : new Date()});
            },
            User: () => {

                return {
                    'user_name' : 'some',
                };
            }
        });
    }

    render() {

        if( this.state.reload ){

            setTimeout(() => {

                this.setState({'reload' : null});

            }, 1);

            return (
                <View style={styles.container}>
                    <Loader/>
                </View>
            );
        }
        if( this.state.load ){

            return (
                <View style={styles.container}>
                    <Loader/>
                </View>
            );
        }

        console.log('render');

        return (
            <View style={styles.container}>
                <ControllerIndex method="Index"/>
                <View>
                    <ControllerIndex method="Footer"/>
                </View>
                <Popup.ShowAlert/>
                <Popup.ShowFullAlert/>
                <Popup.ShowBottomAlert/>
                <Popup.ShowTopAlert/>
                <Popup.ShowModal/>
                <Popup.ShowBottomModal/>
                <Popup.ShowLoader/>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
});

test('renders correctly', () => {
    //let tree = Test('see');
    //console.log(tree);
    const tree = renderer.create(<MainApp />).toJSON();
    expect(tree).toMatchSnapshot();
});
