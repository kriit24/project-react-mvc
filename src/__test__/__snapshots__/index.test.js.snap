// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
          "justifyContent": "center",
        },
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
        {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <ActivityIndicator
      color="#1976d2"
      size="large"
    />
  </View>
</View>
`;
